hidimstat/scenario.py

```diff
@@ -182,3 +182,58 @@ def multivariate_simulation(n_samples=100,
         return X, y, beta, epsilon, X_, w
 
     return X, y, beta, epsilon
+
+
+def multivariate_temporal_simulation(n_samples=100, n_features=500,
+                                     n_targets=30, support_size=10,
+                                     sigma=1.0, rho=0.0, seed=0):
+    """Generate 1D temporal data with constant design matrix
+
+    Parameters
+    -----------
+    n_samples : int
+        Number of samples
+    n_features : int
+        Number of features
+    n_targets : int
+        Number of time points
+    support_size: int
+        Size of the row support
+    sigma : float
+        Standard deviation of the noise at each time point
+    rho : float
+        Level of autocorrelation in the noise
+    seed : int
+        Seed used for generating design matrix and noise
+
+    Returns
+    -------
+    X: ndarray, shape (n_samples, n_features)
+        Design matrix
+    Y : ndarray, shape (n_samples, n_targets)
+        Target
+    Beta : ndarray, shape (n_features, n_targets)
+        Parameter matrix
+    E : ndarray, shape (n_samples, n_targets)
+        Noise matrix
+    """
+
+    rng = np.random.default_rng(seed)
+
+    X = rng.standard_normal((n_samples, n_features))
+
+    Beta = np.zeros((n_features, n_targets))
+    Beta[0:support_size, :] = 1.0
+
+    E = np.zeros((n_samples, n_targets))
+    E[:, 0] = rng.standard_normal(n_samples)
+
+    for i in np.arange(1, n_targets):
```

Comment:
This code seems quite inefficient: it seems like a manual filtering operation.

Reply:
The advantage of this method is that it controls the level of autocorrelation precisely while with Gaussian filtering it is not possible. And, I did not find any shortcut to do that (neither in numpy or scipy).